src/main/java/org/mule/extension/api/annotation/metadata/MetadataScope.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.annotation.metadata;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import org.mule.extension.api.annotation.Extension;
+import org.mule.extension.api.metadata.MetadataResolver;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates the {@link MetadataResolver} to be used to resolve MetaData of a particular Operation.
+ * This annotation can be used at Operation or {@link Extension} level.
+ *
+ * @since 1.0
+ */
+@Target({METHOD, TYPE})
+@Retention(RUNTIME)
+@Documented
+public @interface MetadataScope
+{
+
+    /**
+     * The class within the {@link #value()} must implement {@link MetadataResolver}.
```

Comment:
it is redundant to say that the class must implement MetadataResolver or it won't compile. The method signature already implies that and Java itself won't compile otherwise. Simply say that it returns the class of the metadata resolver to be used